packages/protocol/test/governance/double_signing_slasher.ts

```diff
@@ -0,0 +1,217 @@
+    await registry.setAddressFor(CeloContractName.Validators, validators.address)
+    await registry.setAddressFor(CeloContractName.DoubleSigningSlasher, slasher.address)
+    await registry.setAddressFor(CeloContractName.DowntimeSlasher, accounts[5])
+    await registry.setAddressFor(CeloContractName.GovernanceSlasher, accounts[6])
+    await registry.setAddressFor(CeloContractName.Governance, accounts[7])
+    await validators.initialize(
+      registry.address,
+      groupLockedGoldRequirements.value,
+      groupLockedGoldRequirements.duration,
+      validatorLockedGoldRequirements.value,
+      validatorLockedGoldRequirements.duration,
+      validatorScoreParameters.exponent,
+      validatorScoreParameters.adjustmentSpeed,
+      membershipHistoryLength,
+      maxGroupSize
+    )
+    const group = accounts[0]
+    await registerValidatorGroupWithMembers(group, [validator])
+    await registerValidatorGroupWithMembers(accounts[3], [accounts[4]])
+    await slasher.initialize(registry.address, 10000, 100)
+    await mockLockedGold.incrementNonvotingAccountBalance(validator, 50000)
+    await mockLockedGold.incrementNonvotingAccountBalance(group, 60000)
+  })
+
+  describe('#initialize()', () => {
+    it('should have set the owner', async () => {
+      const owner: string = await slasher.owner()
+      assert.equal(owner, accounts[0])
+    })
+    it('should have set slashing incentives', async () => {
+      const res = await slasher.slashingIncentives()
+      assert.equal(res[0].toNumber(), 10000)
+      assert.equal(res[1].toNumber(), 100)
+    })
+  })
+
+  describe('#setSlashingIncentives()', () => {
+    it('can only be set by the owner', async () => {
+      await assertRevert(slasher.setSlashingIncentives(123, 67, { from: nonOwner }))
+    })
+    it('should have set slashing incentives', async () => {
+      await slasher.setSlashingIncentives(123, 67)
+      const res = await slasher.slashingIncentives()
+      assert.equal(res[0].toNumber(), 123)
+      assert.equal(res[1].toNumber(), 67)
+    })
+  })
+
+  describe('#slash()', () => {
+    let blockNumber: number
+    const validatorIndex = 23
+    const blockA = ['0x12', '0x12', '0x12']
+    const blockB = ['0x13', '0x13', '0x13']
+    const blockC = ['0x11', '0x13', '0x14']
+    beforeEach(async () => {
+      blockNumber = await web3.eth.getBlockNumber()
+      const block = await web3.eth.getBlock(blockNumber)
+      const hash = block.parentHash
+      await slasher.setParentHashFromHeader(blockA, hash)
+      await slasher.setParentHashFromHeader(blockB, '0x34')
+      await slasher.setParentHashFromHeader(blockC, hash)
+      await slasher.setEpochSigner(Math.floor(blockNumber / EPOCH), validatorIndex, validator)
+      await slasher.setEpochSigner(Math.floor(blockNumber / EPOCH), validatorIndex + 1, validator)
+      const bitmap = '0x0000000000000000000000000000000000000000000000000000000000800001'
```

Comment:
What is this used for? Can you include a comment explaining what this bitmap represents? (i.e. downtime of validator X, double signing by validator Y)